### PR TITLE
Update user guide for namespace label in Katib Python SDK example

### PR DESCRIPTION
**Pull request by Doris-xm**
**What this PR does:**

Before creating Katib Experiments, the label for metrics collector injection `katib.kubeflow.org/metrics-collector-injection: enabled` should be presented in the namespace. 

This PR updates the user guide to inform users of the effect of this label.

**Which issue(s) this PR fixes:**
Fixes https://github.com/kubeflow/katib/issues/2499

### CODE REVIEW

File: content/en/docs/components/katib/getting-started.md [hunk 1]
```diff
@@ -17,6 +17,31 @@ You need to install the following Katib components to run examples:
 
 ## Getting Started with Katib Python SDK
 
+### Adding Namespace Label for the Metrics Collector Injection
```
Comment (andreyvelich) [inline content/en/docs/components/katib/getting-started.md:20]: I would suggest that we add this section to the Katib Metrics Collector guide for now: https://www.kubeflow.org/docs/components/katib/user-guides/metrics-collector/
The getting started example, tell users to run Experiment in `kubeflow` namespace, which has the required label in default standalone installation.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this @Doris-xm!
/assign @hbelmiro @varodrig @helenxie-bit @Electronic-Waste

File: content/en/docs/components/katib/getting-started.md (hunk 1 above)
Comment (helenxie-bit) [reply content/en/docs/components/katib/getting-started.md:20]: I noticed something strange. When I previously ran the get started example with the `kubeflow` namespace, I [encountered the "namespace lacks label" error](https://github.com/kubeflow/katib/issues/2492). However, I just tried it again, and now it works fine with the `kubeflow` namespace.

On the other hand, when I use the API to run LLM hyperparameter optimization using `kubeflow` namespace too, the "namespace lacks label" error still occurs.

@andreyvelich Do you have any idea what might be causing this issue? I’ll continue investigating as well.

Comment (andreyvelich) [reply content/en/docs/components/katib/getting-started.md:20]: How did you install Katib and Training Operator when you run LLM hyperparameter optimization using `kubeflow` namespace?

Comment (helenxie-bit) [reply content/en/docs/components/katib/getting-started.md:20]: I used the following commands from the installation page to install the latest changes of Katib control plane and Training Operator control plane:
`kubectl apply -k "github.com/kubeflow/katib.git/manifests/v1beta1/installs/katib-standalone?ref=master"`

`kubectl apply --server-side -k "github.com/kubeflow/training-operator.git/manifests/overlays/standalone?ref=master"`

Comment (andreyvelich) [reply content/en/docs/components/katib/getting-started.md:20]: So, when you deploy Training Operator after the Katib, is might override the namespace label, since it doesn't have it: https://github.com/kubeflow/training-operator/blob/master/manifests/overlays/standalone/namespace.yaml
You should install Training Operator before Katib ideally.
Also, to use PyTorchJob CRD within Katib Trial, it needs to be exist before Katib Controller starts: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/trial/trial_controller.go#L121-L132

Comment (helenxie-bit) [reply content/en/docs/components/katib/getting-started.md:20]: Oh I see! Problem fixed! Thank you so much

Comment (Electronic-Waste) [reply content/en/docs/components/katib/getting-started.md:20]: > I would suggest that we add this section to the Katib Metrics Collector guide for now: https://www.kubeflow.org/docs/components/katib/user-guides/metrics-collector/
The getting started example, tell users to run Experiment in kubeflow namespace, which has the required label in default standalone installation.

SGTM.

File: content/en/docs/components/katib/getting-started.md [hunk 2]
```diff
@@ -17,6 +17,31 @@ You need to install the following Katib components to run examples:
...
+
+Before running your hyperparameter tuning Katib Experiment with Python SDK, 
+ensure the namespace label `katib.kubeflow.org/metrics-collector-injection: enabled` 
+is present. This label enables the sidecar container injection to collect metrics during the experiment.
+
+You can configure the namespace using the following YAML:
```
Comment (varodrig) [inline content/en/docs/components/katib/getting-started.md:26]: Since the user needs to update the namespace, no create a new one. I'd update the sentence by
"You can configure the namespace by adding the following label  katib.kubeflow.org/metrics-collector-injection: enabled as is shown in the sample code."

File: content/en/docs/components/katib/getting-started.md (hunk 1 above)
Comment (Electronic-Waste) [reply content/en/docs/components/katib/getting-started.md:20]: > @andreyvelich @Electronic-Waste should we add a pre-req section on https://www.kubeflow.org/docs/components/katib/user-guides/metrics-collector/ to include this step?

SGTM. We could also declare that `kubeflow` namespace has already been configured with this label, that's why they can use `tune` function in `kubeflow` namespace without any prerequisite setup.

Such as "If you want to use pull-based metrics collector, you need to attach the label `xxx` to the target namespace (`kubeflow` namespace has built-in support for this label)"

Comment (Doris-xm) [reply content/en/docs/components/katib/getting-started.md]: Thanks for your review. I've updated the doc:
- move the detailed guidance to Metrics Collector
- add a **Note** in Getting Started to explain `kubeflow` namespace has built-in support. I also include a link to Metrics Collector for detailed instruments.

Please let me know if you have any suggestions.

File: content/en/docs/components/katib/user-guides/metrics-collector.md [hunk 3]
```diff
@@ -6,6 +6,30 @@ weight = 40
 
 This guide describes how Katib metrics collector works.
 
+## Prerequisites
+
+Before running your hyperparameter tuning Katib Experiment with Python SDK,
+ensure the namespace label `katib.kubeflow.org/metrics-collector-injection: enabled`
+is present. This label enables the sidecar container injection to collect metrics during the experiment.
```
Comment (Electronic-Waste) [inline content/en/docs/components/katib/user-guides/metrics-collector.md:13]:
```suggestion
is present. This label enables the sidecar container injection for pull-based metrics collectors to collect metrics during the experiment.
```

**Review by Electronic-Waste — COMMENTED** [review]
Basically LGTM! Just one comment for you @Doris-xm!

**Review by Electronic-Waste — APPROVED** [review]
LGTM! Thanks for your great contribution @Doris-xm!

/lgtm
/assign @kubeflow/wg-automl-leads @helenxie-bit

**Comment by helenxie-bit** [comment]
Thanks for the contribution!

/lgtm

**Review by andreyvelich — COMMENTED** [review]
Thank you for this contribution @Doris-xm!
/lgtm
/approve
